Sources/NWWebSocket/Model/Client/NWWebSocket.swift

```diff
@@ -244,14 +245,17 @@ open class NWWebSocket: WebSocketConnection {
     /// or a `NWError` if the migration failed for some reason.
     private func migrateConnection(completionHandler: @escaping (Result<WebSocketConnection, NWError>) -> Void) {
 
-        let migratedConnection = NWConnection(to: endpoint, using: parameters)
+        migratedConnection?.cancel()
```

Comment:
if migrateConnection was already called and successfully ended, in the next call when you cancelling migratedConnection you cancelling current 'connection'